### DISCUSSION TITLE
Why does a string array return `string[] | undefined` with a default value?

### DISCUSSION THREAD

**Discussion by indifferentghost** [open-circle/valibot#460] [Q&A] [answered]
I am parsing an object and I'm hoping to have the typescript return with a default value, at the moment I'm receiving an `undefined | string[]`, but it should be returning with the default, below is the only one I've found that works, but it should be easier.

```ts
import {
	array,
	fallback,
	optional,
	parse,
	string,
} from "valibot/mod.ts";

const refreshInit = optional(object({
	// ...
	ignoreKinds: fallback(array(string()), ["any", "access"] as const),
	// ...
}), {ignoreKinds: ['any', 'access']});
```

I have tried the following, which seems to work on primitive values.
```ts
	ignoreKinds: optional(array(string()), ["any", "access"] as const),
```

**Comment by fabian-hiller** [comment, marked as answer]
Unfortunately, I don't quite understand what you mean. Everything works as it should for me. Can you explain it in more detail?

```ts
import * as v from 'valibot';

const Schema = v.optional(
  v.object({
    key: v.array(v.string()),
  }),
  { key: ['foo'] }
);

type Input = v.Input<typeof Schema>; // { key: string[] } | undefined
type Output = v.Output<typeof Schema>; // { key: string[] }
```

```ts
import * as v from 'valibot';

const Schema = v.object({
  key: v.optional(v.array(v.string()), ['foo']),
});

type Input = v.Input<typeof Schema>; // { key?: string[] | undefined }
type Output = v.Output<typeof Schema>; // { key: string[] }
```

**Reply by indifferentghost** [reply]
I suppose I was expecting `['foo'] as const` to work and it didn't. https://replit.com/@ThomasD/DeadlyEveryCallback#index.ts

Removing `as const` has it work fine.
